test/python/unit/api/test_datatype_api.py

```diff
@@ -294,3 +294,249 @@ def test_parametric_datatype(solver):
     assert not pairIntReal.isSubsortOf(pairIntInt)
     assert not pairRealInt.isSubsortOf(pairIntInt)
     assert pairIntInt.isSubsortOf(pairIntInt)
+
+
+def test_datatype_simply_rec(solver):
+    # Create mutual datatypes corresponding to this definition block:
+    #
+    #   DATATYPE
+    #     wlist = leaf(data: list),
+    #     list = cons(car: wlist, cdr: list) | nil,
+    #     ns = elem(ndata: set(wlist)) | elemArray(ndata2: array(list, list))
+    #   END
+
+    # Make unresolved types as placeholders
+    unresTypes = set([])
+    unresWList = solver.mkUninterpretedSort("wlist")
+    unresList = solver.mkUninterpretedSort("list")
+    unresNs = solver.mkUninterpretedSort("ns")
+    unresTypes.add(unresWList)
+    unresTypes.add(unresList)
+    unresTypes.add(unresNs)
+
+    wlist = solver.mkDatatypeDecl("wlist")
+    leaf = solver.mkDatatypeConstructorDecl("leaf")
+    leaf.addSelector("data", unresList)
+    wlist.addConstructor(leaf)
+
+    llist = solver.mkDatatypeDecl("list")
+    cons = solver.mkDatatypeConstructorDecl("cons")
+    cons.addSelector("car", unresWList)
+    cons.addSelector("cdr", unresList)
+    llist.addConstructor(cons)
+    nil = solver.mkDatatypeConstructorDecl("nil")
+    llist.addConstructor(nil)
+
+    ns = solver.mkDatatypeDecl("ns")
+    elem = solver.mkDatatypeConstructorDecl("elem")
+    elem.addSelector("ndata", solver.mkSetSort(unresWList))
+    ns.addConstructor(elem)
+    elemArray = solver.mkDatatypeConstructorDecl("elemArray")
+    elemArray.addSelector("ndata", solver.mkArraySort(unresList, unresList))
+    ns.addConstructor(elemArray)
+
+    dtdecls = []
+    dtdecls.append(wlist)
+    dtdecls.append(llist)
+    dtdecls.append(ns)
```

Comment:
```suggestion
    dtdecls = [wlist, llist, ns]
```

(yeah, the corresponding C++ test pushes them separately but I don't think that it adds much here)